packages/lab/src/header-block/HeaderBlock.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  H2,
+  H3,
+  StatusIndicator,
+  Text,
+  type ValidationStatus,
+  makePrefixer,
+  useDialogContext,
+} from "@salt-ds/core";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import clsx from "clsx";
+import { type ComponentPropsWithoutRef, forwardRef } from "react";
+import headerBlockCss from "./HeaderBlock.css";
+
+import { HeaderBlockCloseButton } from "./HeaderBlockCloseButton";
+
+export type HeaderBlockPadding = "100" | "200" | "300";
+
+const withBaseName = makePrefixer("saltHeaderBlock");
+
+export interface HeaderBlockProps extends ComponentPropsWithoutRef<"div"> {
+  /**
+   * The status of the Dialog
+   */
+  status?: ValidationStatus | undefined;
+  /**
+   * Displays the accent bar in the Dialog Title */
+  accent?: boolean;
+  /**
+   * Header text
+   */
+  header?: string;
+  /**
+   * Preheader text is displayed just above the header
+   **/
+  preheader?: string;
+  /**
+   * Description text is displayed just below the header
+   **/
+  description?: string;
+  /**
+   * Padding of the header block. Default is "300"
+   **/
+  padding?: HeaderBlockPadding;
```

Comment:
This is the first introduction of spacing to any component. Worth a wider discussion around options / syntax (e.g., v.s. `size` in Icon / `gap` in layout component) being multiplier.

Reply:
IIRC there was a lot of discussion about this with @navkaur76 and @dplsek, so they might be able to provide more context.

To take the `gap` example as a pattern, this could be switched to `padding: 1 | 2 | 3`?